src/inference/gibbs.jl

```diff
@@ -29,17 +29,33 @@ methods like Particle Gibbs. You can increase the effectiveness of particle samp
 more particles in the particle sampler.
 """
 mutable struct Gibbs{A} <: InferenceAlgorithm
-    n_iters   ::  Int     # number of Gibbs iterations
     algs      ::  A   # component sampling algorithms
-    thin      ::  Bool    # if thinning to output only after a whole Gibbs sweep
+    function Gibbs(algs...)
+        algs = algs isa Tuple ? algs : (algs, )
+        return new{typeof(algs)}(algs)
+    end
+end
+
+alg_str(::Sampler{<:Gibbs}) = "Gibbs"
+transition_type(spl::Sampler{<:Gibbs}) = typeof(transition(spl))
+
+mutable struct GibbsState{V<:VarInfo, S<:Tuple{Vararg{Sampler}}, T<:NamedTuple} <: AbstractSamplerState
+    vi::V
+    samplers::S
+    subsamples::T
+end
+
+function GibbsState(model::Model, samplers::S) where S<:Tuple{Vararg{Sampler}}
+    ids = tuple([Symbol(s.selector.gid) for s in samplers]...)
+    refs = tuple([Ref{transition_type(s)}() for s in samplers]...)
+    nt = NamedTuple{ids}(refs)
+    return GibbsState(VarInfo(model), samplers, nt)
 end
-Gibbs(n_iters::Int, algs...; thin=true) = Gibbs(n_iters, algs, thin)
 
 const GibbsComponent = Union{Hamiltonian,MH,PG}
 
 function Sampler(alg::Gibbs, model::Model, s::Selector)
```

Comment:
Maybe we can improve the naming in the Gibbs sampler a bit. Currently `samples` can refer to sub samplers while `spl` can refer to the Gibbs, which is a bit hard to read.

Maybe we can enforce 
- `spl`/`alg` for the Gibbs
- `sub_spl`/`sub_alg` for the compoenets?

PS: open for other names

Reply:
I think it most places we use `local_spl`, which I think is a reasonable convention. Maybe `sub_spl` would be better though.

Reply:
Both are fine to me.